docs/testing/DEVELOPER-BEST-PRACTICES.md

```diff
@@ -0,0 +1,90 @@
+# Developer Best Practices
```

Comment:
This file is called developer best practices but it seems to contain - something about go, a set of instructions on installing and configuring git, and then are we really telling users how we suggest that they work with git within the Appsody project (without actually saying that)? What is the intent of this information - it isn't what I think of as "best practices"?